lib/vrl/lookup/src/lookup_v2/mod.rs

```diff
@@ -1,51 +1,83 @@
-//! This crate contains all of the logic for lookup paths.
+//! This crate contains all of the logic for paths.
+//!
+//! Paths can be thought of as similar to file paths (in an operating system) pointing
+//! to specific files inside of a directory.
+//! A `Value` is a data structure that can contain recursively nested fields. Paths
+//! allow referring to a specific field inside of a `Value`.
+//!
+//! # Example
+//! Below is a sample `Value`. Different fields can be accessed with paths.
+//! ```json
+//! {
+//!   "foo": {
+//!       "bar": 1
+//!    },
+//!    "baz": ["a", "b", "c"]
+//! }
+//! ```
+//!
+//! | path   | value it points to                    |
+//! |--------|---------------------------------------|
+//! | `.foo.bar` | `1`
+//! | `.foo` | `{ "bar": 1 }`
+//! | `.`    | `{ "foo" : { "bar" : 1 }, "baz" : ["a", "b", "c"] }`
+//! | `.baz[0]` | `"a"`
+//! | `.baz` | `["a", "b", "c"]`
+//!
 //!
 //! # Traits
-//! There are 2 main traits that define a path.
-//! - [ValuePath] is a path that points inside of a `Value`
-//! - [TargetPath] is a path that points to a `target`. A `target` in VRL refers to
-//! the external data being processed by a VRL script. A `target` has 2 main sections that can be
+//! There are 2 main traits that define a path. Most functions that use a path for querying
+//! will require one of these traits, rather than a concrete type.
+//!
+//! - [ValuePath] is a path that points to a field inside of a `Value`.
+//! - [TargetPath] is a path that points to a field inside of a `target`. A `target` in VRL refers to
+//! the external data being processed by a VRL script. A `target` has two main sections that can be
 //! pointed to, `event` and `metadata`.  [TargetPath::prefix] identifies the section, and
 //! [TargetPath::value_path] is a [ValuePath] pointing into that section.
 //!
 //! Note that for performance reasons, since [ValuePath] and [TargetPath] require [Clone], these
 //! traits are only implemented on types that are cheap to clone (usually references). That means
 //! when passing in a value (e.g. [OwnedValuePath]) into a function that requires `impl ValuePath`,
```

Comment:
```suggestion
//! when passing a value (e.g. [OwnedValuePath]) into a function that requires `impl ValuePath`,
```

missed this the first time